tools/code_format/check_format.py

```diff
@@ -916,25 +925,25 @@ def checkFormat(file_path):
   error_messages = []
   # Apply fixes first, if asked, and then run checks. If we wind up attempting to fix
   # an issue, but there's still an error, that's a problem.
-  try_to_fix = operation_type == "fix"
+  try_to_fix = command_line_args.operation_type == "fix"
   if isBuildFile(file_path) or isSkylarkFile(file_path) or isWorkspaceFile(file_path):
     if try_to_fix:
-      error_messages += fixBuildPath(file_path)
-    error_messages += checkBuildPath(file_path)
+      error_messages += fixBuildPath(file_path, command_line_args)
+    error_messages += checkBuildPath(file_path, command_line_args)
   else:
     if try_to_fix:
-      error_messages += fixSourcePath(file_path)
-    error_messages += checkSourcePath(file_path)
+      error_messages += fixSourcePath(file_path, command_line_args)
+    error_messages += checkSourcePath(file_path, command_line_args)
 
   if error_messages:
     return ["From %s" % file_path] + error_messages
   return error_messages
 
 
-def checkFormatReturnTraceOnError(file_path):
+def checkFormatReturnTraceOnError(file_path, command_line_args):
```

Comment:
What do you think about making this a class and have command line args be a member variable? That could reduce the amount of threading required.

Reply:
This sounds like a better approach. I'll get to this this week.

Reply:
Sorry for the delay, this is now ready for another look